aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -43,34 +57,45 @@ async def sign(request: web.BaseRequest):
 
     """
     response = {}
+    body = await request.json()
+    doc = body.get("doc")
     try:
         context: AdminRequestContext = request["context"]
-        body = await request.json()
-        verkey = body.get("verkey")
-        doc = body.get("doc")
-        credential = doc["credential"]
-        signature_options = doc["options"]
-
         async with context.session() as session:
-            wallet = session.inject(BaseWallet, required=False)
-            if not wallet:
-                raise web.HTTPForbidden(reason="No wallet available")
-            document_with_proof = await sign_credential(
-                credential, signature_options, verkey, wallet
+            doc_with_proof = await sign_credential(
+                session, doc.get("credential"), doc.get("options"), body.get("verkey")
             )
-
-        response["signed_doc"] = document_with_proof
-    except (DroppedAttributeError, MissingVerificationMethodError) as err:
-        response["error"] = str(err)
-
+            response["signed_doc"] = doc_with_proof
+    except (BaseJSONLDMessagingError) as err:
+        response["error"] = repr(err)
```

Comment:
repr likely isn't returning the sort of information we want to be reporting back in the response. I think `str` was probably the correct behavior here.

Reply:
repr is really more for development and not for reporting. I will change back to str.